tests/bluetooth/tester/src/btp_cap.c

```diff
@@ -0,0 +1,1028 @@
+
+	uint8_t conn_index;
+
+	if (err != 0) {
+		LOG_DBG("discover failed (%d)", err);
+		return;
+	}
+
+	if (set_count == 0) {
+		LOG_DBG("Device has no sets");
+		return;
+	}
+
+	conn_index = bt_conn_index(conn);
+
+	LOG_DBG("Found %zu sets on member[%u]", set_count, conn_index);
+
+	for (size_t i = 0U; i < set_count; i++) {
+		LOG_DBG("CSIS[%zu]: %p", i, &member->insts[i]);
+		LOG_DBG("Rank: %u", member->insts[i].info.rank);
+		LOG_DBG("Set Size: %u", member->insts[i].info.set_size);
+		LOG_DBG("Lockable: %u", member->insts[i].info.lockable);
+	}
+
+	cur_csis_inst = &((struct bt_csip_set_coordinator_set_member *)member)->insts[0];
+	set_members[conn_index] = member;
+}
+
+static void csip_lock_changed_cb(struct bt_csip_set_coordinator_csis_inst *inst,
+				 bool locked)
+{
+	LOG_DBG("");
+}
+
+static void csip_set_coordinator_ordered_access_cb(
+	const struct bt_csip_set_coordinator_set_info *set_info, int err,
+	bool locked,  struct bt_csip_set_coordinator_set_member *member)
+{
+	LOG_DBG("");
+
+	if (err) {
+		LOG_ERR("Ordered access failed with err %d", err);
+	} else if (locked) {
+		LOG_DBG("Ordered access procedure locked member %p", member);
+	} else {
+		LOG_DBG("Ordered access procedure finished");
+	}
+}
+
+static struct bt_csip_set_coordinator_cb set_coordinator_cbs = {
+	.lock_set = csip_set_coordinator_lock_set_cb,
+	.release_set = csip_set_coordinator_lock_release_cb,
+	.discover = csip_discover_cb,
+	.lock_changed = csip_lock_changed_cb,
+	.ordered_access = csip_set_coordinator_ordered_access_cb
+};
+
+static bool csip_set_coordinator_oap_cb(const struct bt_csip_set_coordinator_set_info *set_info,
+					struct bt_csip_set_coordinator_set_member *members[],
+					size_t count)
+{
+	for (size_t i = 0; i < count; i++) {
+		LOG_DBG("Ordered access for members[%zu]: %p", i, members[i]);
+	}
+
+	return true;
+}
+
+static uint8_t btp_cap_ordered_access(const void *cmd, uint16_t cmd_len,
+				      void *rsp, uint16_t *rsp_len)
+{
+	const struct bt_csip_set_coordinator_set_member *members[ARRAY_SIZE(set_members)];
+	const struct btp_cap_ordered_access_cmd *cp = cmd;
+	unsigned long member_count = 0;
+	struct bt_conn *conn;
+	int err;
+
+	LOG_DBG("");
+
+	conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, &cp->address);
+	if (!conn) {
+		LOG_ERR("Unknown connection");
+
+		return BTP_STATUS_FAILED;
+	}
+	bt_conn_unref(conn);
+
+	if (cur_csis_inst == NULL) {
+		LOG_ERR("No CISP instance available");
+
+		return BTP_STATUS_FAILED;
+	}
+
+	for (size_t i = 0; i < (size_t)ARRAY_SIZE(set_members); i++) {
+		if (set_members[i] == NULL) {
+			continue;
+		}
+
+		members[member_count++] = set_members[i];
+	}
+
+	if (member_count == 0) {
+		LOG_ERR("No set members available");
+
+		return BTP_STATUS_FAILED;
+	}
+
+	err = bt_csip_set_coordinator_ordered_access(members,
+						     member_count,
+						     &cur_csis_inst->info,
+						     csip_set_coordinator_oap_cb);
+	if (err != 0) {
+		return BTP_STATUS_FAILED;
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
```

Comment:
Yeah, unfortunately the CAP procedures are not fully working for CSIP yet, as this should be done by the stack when providing a CSIP set. 

For now this is fine, and ideally this can be (or should be possible) to remove once CAP support CSIP properly.

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/67104